src/Raven.Server/Documents/Sharding/ShardedContext.cs

```diff
@@ -114,5 +131,114 @@ public bool HasClientConfigurationChanged(long clientConfigurationEtag)
             var actual = Hashing.Combine(lastClientConfigurationIndex, _lastClientConfigurationIndex);
             return actual > clientConfigurationEtag;
         }
+
+        public string GetShardedDatabaseName(int index = 0)
+        {
+            if (index >= _record.Shards.Length)
+                throw new InvalidOperationException($"Requested shard '{index}' of database '{DatabaseName}' but shards length '{_record.Shards.Length}'.");
+
+            return _record.DatabaseName + "$" + index;
+        }
+
+        public List<string> GetShardedDatabaseNames()
+        {
+            var list = new List<string>();
+            for (int i = 0; i < Count; i++)
+            {
+                list.Add(GetShardedDatabaseName(i));
+            }
+            return list;
+        }
+
+        public async Task<string> GetLastDocumentChangeVectorForCollection(string subCollection)
```

Comment:
What is this for? And what is `subCollection` ?

Reply:
this is used if the subscription is created with `SubscriptionChangeVectorSpecialStates.LastDocument` option to retrieve the latest change vector from all shards, `subCollection` is the subscription collection